File: aoc/utils/typescript/submit.ts
import { type Day, parseSolveCliArgs, type Part, type Year } from './cli.ts'
import { type Answer, get_answer, isInteger } from './run.ts'

/**
 * See: https://docs.deno.com/api/deno/~/Deno.Command
 */
function submitAnswer(year: Year, day: Day, part: Part, answer: Answer) {
  if (!isInteger(answer)) {
    console.log('🤔 Solution is not an integer. Aborting.')
    console.log(`🔍 Answer: ${answer}`)
    return
  }

  console.log('Submitting answer...')

  const aocSessionCookieFile = `${Deno.cwd()}/.aoc-session-cookie`

  // see: https://github.com/scarvalhojr/aoc-cli?tab=readme-ov-file#usage-%EF%B8%8F
  const command = new Deno.Command(Deno.execPath(), {
    args: `aoc s -y ${year} -d ${day} -s ${aocSessionCookieFile} ${part} ${answer}`.split(' '),
  })

  command.outputSync()
  // const { code, stdout, stderr } = command.outputSync()
}

async function main() {
  const args = parseSolveCliArgs()
  const answer = await get_answer(args.year, args.day, args.part)

  submitAnswer(args.year, args.day, args.part, answer)
}

if (import.meta.main) {
  await main()
}

File: aoc/utils/python/submit.py
import subprocess
from pathlib import Path

from rich import print

from aoc.utils.python.cli import Day, Part, Year, parse_solve_cli_args
from aoc.utils.python.run import Answer, get_answer


def submit_answer(year: Year, day: Day, part: Part, answer: Answer) -> None:
    if not isinstance(answer, int):
        print("🤔 Solution is not an integer. Aborting.")
        print(f"🔍 Answer: {answer}")
        return

    aoc_session_cookie_file = Path(".aoc-session-cookie").resolve()

    # see: https://github.com/scarvalhojr/aoc-cli?tab=readme-ov-file#usage-%EF%B8%8F
    command = f"aoc s -y {year} -d {day} -s {aoc_session_cookie_file} {part} {answer}"

    try:
        subprocess.run(command.split(" "), check=True)
    except subprocess.CalledProcessError as e:
        print(f"🛑 Error submitting answer: {e}")
        exit(1)


def main() -> None:
    args = parse_solve_cli_args()
    answer = get_answer(args.year, args.day, args.part)
    submit_answer(args.year, args.day, args.part, answer)


if __name__ == "__main__":
    main()
